content/blog/ggplot2-3-5-0/index.Rmd

```diff
@@ -0,0 +1,385 @@
+pie + coord_radial(start = 0.25 * pi, end = 1.75 * pi, inner.radius = 0.3)
+```
+
+Another drawback about polar coordinates is that it can become pretty involved to coordinate the angles of text annotations with the theta (angle) position of the plot.
+To remove some of this hassle, `coord_radial()` has a `rotate_angle` switch, that will line up the text's `angle` aesthetic with the theta coordinate.
+For text angles of 0 degrees, this will place text in a tangent orientation to the circle and for angles of 90 degrees, this places text along the radius, as in the plot below.
+
+```{r text_angles}
+#| fig.alt = "A wind rose plot showing miles per gallon for different cars. The
+#|  car names skirt the outer edge of the plot and are oriented towards the
+#|  centre."
+ggplot(mtcars, aes(seq_along(mpg), mpg)) +
+  geom_col(width = 1) +
+  geom_text(
+    aes(y = 32, label = rownames(mtcars)),
+    angle = 90, hjust = 1
+  ) +
+  coord_radial(rotate_angle = TRUE, expand = FALSE)
+```
+
+Because the logic of drawing the axes for polar coordinates is not the same as when the axes are vertical or horizontal, we used the new guide system to build an axis specific to `coord_radial()`: the `guide_axis_theta()` axis.
+Guides for `coord_radial()` can be set using the `theta` and `r` name in the `guides()` function, where the `r` guide can be a regular axis.
+The theta guide obeys the theme settings of `axis.{setting}.x.bottom` to display the axis, regardless of whether `theta = "x"` or `theta = "y"` in `coord_radial()`. The primary r-axis likewise uses the `axis.{setting}.y` setting.
+When setting the `angle` parameter in the guide function, these are considered 'relative' to the coordinates.
+Setting `angle = 0` will align the `r`-axis text with the tick marks and the `theta`-axis following the tangent.
+
+```{r axis_styling}
+#| fig.alt = "A pacman-shaped plot with an open quadrant at the top. An axis
+#|  with blue text following a 45 degree angle is displayed in the open 
+#|  quadrant. An axis with long tick marks and red text follows the circle
+#|  sector on the outside, with text angles following the angle of the plot."
+ggplot(mpg, aes(class, displ)) +
+  geom_boxplot() +
+  coord_radial(start = 0.25 * pi, end = 1.75 * pi) +
+  guides(
+    theta = guide_axis_theta(angle = 0),
+    r     = guide_axis(angle = 0)
+  ) +
+  theme(
+    axis.line.x.bottom = element_line(),
+    axis.ticks.length.x.bottom = unit(10, "pt"),
+    axis.text.x.bottom = element_text(color = "red"),
+    axis.text.y.left = element_text(color = "blue")
+  )
+```
+## Patterns
+
+Patterns were first introduced in the {grid} package, which empowers ggplot2, in R 4.1.0 and were refined in R 4.2.0 to support multiple patterns.
+If your graphics device supported it, theme elements could already be set to patterns, even before this release.
+
+> Note: On Windows machines, the default device in RStudio and in the knittr package is `png()`, which does not support patterns.
```

Comment:
```suggestion
> Note: On Windows machines, the default device in RStudio and in the knitr package is `png()`, which does not support patterns.
```

Would be good to note how to switch knitr devices to something (ragg?) that does support patterns.